### PR TITLE
Use `seed` to randomize require order of test files

### PR DESCRIPTION
**Pull request by halfdan**
Addresses https://groups.google.com/g/elixir-lang-core/c/yjs-zaFvZJc

Looking for feedback on implementation and testing strategy.

- [ ] Update docs for `seed` argument

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit.ex [hunk 1]
```diff
@@ -395,8 +395,9 @@ defmodule ExUnit do
   @doc since: "1.12.0"
   @spec async_run() :: Task.t()
   def async_run() do
+    options = persist_defaults(configuration())
+
```
Comment (halfdan) [inline lib/ex_unit/lib/ex_unit.ex:399]: Had to be outside of `Task.async` to ensure `seed` was set when it's fetched in `test.ex`

File: lib/mix/test/mix/tasks/test_test.exs [hunk 2]
```diff
@@ -57,6 +57,32 @@ defmodule Mix.Tasks.TestTest do
...
+        opts = ["--seed", "5", "--trace"]
+        # Forcing the parallel compiler to operate synchronously
+        envs = [{"ELIXIR_ERL_OPTIONS", "+S 1:1"}]
+        output = mix(["test" | opts], envs)
+
+        matches = Regex.scan(~r"\wTest\s\[.*\]", output)
+        assert matches == [["ATest [test/a_test_stale.exs]"], ["BTest [test/b_test_stale.exs]"]]
+      end)
+    end
+  end
```
Comment (josevalim) [inline lib/mix/test/mix/tasks/test_test.exs:84]: Those tests are rather expensive and they rely on implementation details-ish. I am thinking we could just skip them altogether? Other than that, the impl is great!

Comment (halfdan) [reply lib/mix/test/mix/tasks/test_test.exs:84]: Works for me! Let me remove the test.

**Comment by halfdan** [comment]
@josevalim I considered updating the docs for the `--seed` argument, but now I'm thinking that they are probably fine as is:

```
    * `--seed` - seeds the random number generator used to randomize the order of tests;
      `--seed 0` disables randomization so the tests in a single file will always be ran
      in the same order they were defined in
```

Even with `--seed 0` there's no guarantee the modules are loaded in the same order given the parallelisation in how the modules are loaded.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**